content/telegraf/v1.19/external_plugins/shim.md

```diff
@@ -0,0 +1,71 @@
+---
+title: Use the `execd` shim
+description:
+menu:
+  telegraf_1_19:
+     name: Use the `execd` shim
+     weight: 50
+     parent: External plugins
+---
+
+The _shim_ makes it easy to extract an internal input,
+processor, or output plugin from the main Telegraf repo out to a stand-alone
+repo. This allows anyone to build and run it as a separate app using one of the
+`execd` plugins:
+- [inputs.execd](/plugins/inputs/execd)
+- [processors.execd](/plugins/processors/execd)
+- [outputs.execd](/plugins/outputs/execd)
+
+## Steps to externalize a plugin
+
+1. Move the project to an external repo, it's recommended to preserve the path
+  structure, (but not strictly necessary). eg if your plugin was at
+  `plugins/inputs/cpu`, it's recommended that it also be under `plugins/inputs/cpu`
+  in the new repo. For a further example of what this might look like, take a
+  look at [ssoroka/rand](https://github.com/ssoroka/rand) or
+  [danielnelson/telegraf-plugins](https://github.com/danielnelson/telegraf-plugins)
+1. Copy [main.go](./example/cmd/main.go) into your project under the `cmd` folder.
+  This will be the entrypoint to the plugin when run as a stand-alone program, and
+  it will call the shim code for you to make that happen. It's recommended to
+  have only one plugin per repo, as the shim is not designed to run multiple
+  plugins at the same time (it would vastly complicate things).
+1. Edit the main.go file to import your plugin. Within Telegraf this would have
+  been done in an all.go file, but here we don't split the two apart, and the change
+  just goes in the top of main.go. If you skip this step, your plugin will do nothing.
+  eg: `_ "github.com/me/my-plugin-telegraf/plugins/inputs/cpu"`
+1. Optionally add a [plugin.conf](./example/cmd/plugin.conf) for configuration
```

Comment:
We would ideally want people to do this most of the time. It's listed as optional since you don't _need_ to have a `plugin.conf` for your external plugin but it's a very nice to have.